app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/AllowancesAndDeductions.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.util.yourTaxFreeAmount
+
+import uk.gov.hmrc.tai.model.domain._
+import uk.gov.hmrc.tai.model.domain.calculation.CodingComponent
+
+case class AllowancesAndDeductions(allowances: Seq[CodingComponentPair], deductions: Seq[CodingComponentPair]) {
```

Comment:
Would using an apply method and renaming to AllowanceAndDeductionPairs be a bit more descriptive?

Reply:
I have renamed the class but we are unable to use an apply method due to erasure issues